Bali/src/main/java/OffersServlet.java

```diff
@@ -0,0 +1,58 @@
+import java.io.*;
+import javax.servlet.ServletException;
+import javax.servlet.http.*;
+import com.google.gson.Gson;
+
+
+public class OffersServlet extends HttpServlet {
+    private OfferList offerList = new OfferList();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        response.setContentType("json");
+        int skip = 0;
+        int top = 10;
+        if(request.getParameter("skip") != null){
+            skip = Integer.parseInt(request.getParameter("skip"));
+        }
+        if(request.getParameter("top") != null){
+            top = Integer.parseInt(request.getParameter("top"));
+        }
+        response.getWriter().print((new Gson()
+                .toJson(offerList.getPage(skip, top, (new Gson())
+                        .fromJson(request.getReader().readLine(), OfferFilter.class)))));
+    }
+
+    @Override
+    public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        response.setContentType("json");
+        String[] uri = request.getRequestURI().split("/");
+        switch (uri[2]){
+            case "add": {
+                response.getWriter()
```

Comment:
(style) Much better if you split this into separate statements.  Code "ladders" are barely readable. 
```
String line = request.readLine(0;
Gson gson = new Gson().fromJson(line);
...
```
This is not "I'm an artist, I see things like this")) This is strictly about the maintainability of code.